### PR TITLE
Add some common code from archivist-samples

### PR DESCRIPTION
**Pull request by eccles**
Problem:
Unclear how to use from CLI and sometimes we need to only
create an asset/location if it does not already exist.

Solution:
Added convenience code for argparse and setting up a common Archivist
endpoint.

Signed-off-by: Paul Hewlett <phewlett76@gmail.com>

### CODE REVIEW

File: archivist/assets.py [hunk 1]
```diff
@@ -145,9 +148,11 @@ def create(
         # in the method args will overide...
         newprops = _deepmerge({"behaviours": BEHAVIOURS}, props)
         data = self.__query(newprops, attrs)
-        return self.create_from_data(data, confirm=confirm)
+        return self.create_from_data(data, confirm=confirm, exist_ok=exist_ok)
```
Comment (Serhiy1) [inline archivist/assets.py:151]: I don't really like the name of this arg, I feel like it doesn't adequately describe what it does, how about `verify_unique`?

Comment (eccles) [reply archivist/assets.py:151]: Hmm - maybe. I was taking a cue from similar args to os.makedirs - for example...

**Review by JAG-UK — CHANGES_REQUESTED** [review]

File: archivist/assets.py (hunk 1 above)
Comment (eccles) [reply archivist/assets.py]: removed...

File: archivist/parser.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+            "assets": {
+                "storage_integrity": args.storage_integrity.name,
+                "attributes": {
+                    "arc_namespace": namespace,
+                },
+            },
+            "locations": {
+                "attributes": {
+                    "arc_namespace": namespace,
+                },
```
[the remark this answers is not in the document]
Comment (eccles) [reply archivist/parser.py]: fixed

**Review by JAG-UK — APPROVED** [review]

**Pull request merged**